### PR TITLE
BaSP-TG-111: Tasks with Redux

### PR DESCRIPTION
**Pull request by FefeTroanes**
The web page was updated using Redux for state management. This change maintains the structure and the CRUD funcionality, and make its maintenance easier. In this branch we work on the Tasks section, wich is vinculated with Employees and Projects.
The web still works with the alerts and modals implemented in the last week.
The CRUD funcionality ensures the get, create, edit and delete methods, complemented with the front design style.

### CODE REVIEW

**Review by martinalbrnz — DISMISSED** [review]
Good job!

File: src/Components/Tasks/AddTask/index.jsx [hunk 1]
```diff
@@ -13,38 +13,13 @@ const AddTask = (props) => {
...
-    if (res.status === 201) {
-      alert('Task added successfully');
-      setRefresh('');
-      props.handleClose();
-    } else if (res.status === 400 || res.status === 500) {
-      alert('Something went wrong');
-    }
+    dispatch(thunks.addTask(task));
+    props.handleClose();
+    console.log('NEW TASK', task);
```
Comment (nicolascostanza) [inline src/Components/Tasks/AddTask/index.jsx:21]: este console.log fuera

File: src/Components/Tasks/index.js [hunk 2]
```diff
@@ -61,7 +58,7 @@ function Tasks() {
         setAssignedEmployee(response.data.assignedEmployee);
         setStartDate(response.data.startDate);
         setStatus(response.data.status);
-        console.log('data:', response.data);
+        console.log('edit modal data:', response.data);
```
Comment (nicolascostanza) [inline src/Components/Tasks/index.js:61]: fuera el console.log

File: src/Components/Tasks/index.js [hunk 3]
```diff
@@ -87,33 +84,19 @@ function Tasks() {
...
   };
 
   const deleteTask = async (id) => {
-    const resp = confirm('Are you sure you want to delete this task?');
-    if (resp) {
-      await fetch(`${process.env.REACT_APP_API_URL}/tasks/${id}`, {
-        method: 'DELETE'
-      });
-      setTask(tasks.filter((task) => task._id !== id));
+    const deleteConfirm = confirm('Are you sure you want to delete this task?');
```
Comment (nicolascostanza) [inline src/Components/Tasks/index.js:92]: aca estaria bueno a futuro agregarle modal en vez de un confirm, pero esta correcto

**Review by martinalbrnz — APPROVED** [review]
LGTM!

**Review by nicolascostanza — DISMISSED** [review]

**Comment by RominaRIP** [comment]
Testing Results

Federico and Martín, please check the functionality of the Add feature.

Tested locally on https://alfonso-trackgenix-app-git-tg-111-basp-m2022.vercel.app/tasks
Test Results: Failed ⚠️

Bugs found:

- Task can't be created, please check the 'Add' feature: Once the user clicks on the 'Submit' button in the filled modal form, the page turns blank. If the user refreshes the page, the data of the new task filled in the form doesn't shows in the table.

**Review by martinalbrnz — APPROVED** [review]
Good job!

**Review by nicolascostanza — APPROVED** [review]
atr

**Comment by RominaRIP** [comment]
2nd Testing Results

Federico and Martín, everything works as expected. Good job!

Tested locally on https://alfonso-trackgenix-app-git-tg-111-basp-m2022.vercel.app/tasks
Test Results: Passed ✅

**Pull request merged**